src/main/java/com/arpnetworking/metrics/common/sources/CollectdHttpSourceV1.java

```diff
@@ -0,0 +1,290 @@
+import com.google.common.collect.Maps;
+import net.sf.oval.ConstraintViolation;
+import net.sf.oval.constraint.NotNull;
+import net.sf.oval.exception.ConstraintsViolatedException;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Processes HTTP posts from Collectd, extracts data and emits metrics.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class CollectdHttpSourceV1 extends ActorSource {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected Props createProps() {
+        return Actor.props(this);
+    }
+
+    /**
+     * Protected constructor.
+     *
+     * @param builder Instance of <code>Builder</code>.
+     */
+    protected CollectdHttpSourceV1(final Builder builder) {
+        super(builder);
+        _parser = builder._parser;
+    }
+
+    private final Parser<List<DefaultRecord.Builder>> _parser;
+
+    /**
+     * Internal actor to process requests.
+     */
+    /* package private */ static final class Actor extends UntypedActor {
+        /**
+         * Creates a {@link Props} for this actor.
+         *
+         * @param source The {@link CollectdHttpSourceV1} to send notifications through.
+         * @return A new {@link Props}
+         */
+        /* package private */ static Props props(final CollectdHttpSourceV1 source) {
+            return Props.create(Actor.class, source);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void onReceive(final Object message) throws Exception {
+            if (message instanceof RequestReply) {
+                final RequestReply request = (RequestReply) message;
+                Source.single((HttpRequest) request.getRequest())
+                        .via(_processGraph)
+                        .toMat(_sink, Keep.right())
+                        .run(_materializer)
+                        .whenComplete((done, err) -> {
+                            final CompletableFuture<HttpResponse> responseFuture = request.getResponse();
+                            if (err == null) {
+                                responseFuture.complete(HttpResponse.create().withStatus(200));
+                            } else {
+                                BAD_REQUEST_LOGGER.warn()
+                                        .setMessage("Error handling collectd post")
+                                        .setThrowable(err)
+                                        .log();
+                                if (err instanceof ParsingException) {
+                                    responseFuture.complete(HttpResponse.create().withStatus(400));
+                                } else if (err instanceof ConstraintsViolatedException) {
```

Comment:
I thought that with the exception handling change that these would no longer leak out to here.